run/GCHP/runScriptSamples/gchp.batch_job.sh

```diff
@@ -0,0 +1,157 @@
+# SIMULATION       C90, 2019-01-01 to 2019-01-02
+# RESOURCES        2 nodes, 30 processes per node
+# SEE ALSO         sbatch man pages, http://gchp.rtfd.io/
+#
+
+#################################################################
+#
+# LOAD LIBRARIES - 2 options
+#
+# Option 1: Define environment settings in external file, link to it from
+#           run directory symlink gchp.env, and source it here.
+# 
+#    source gchp.env
+#
+# Option 2: Define environment settings directly in this file. Two
+#           examples are below.
+# 
+#    Example 1: PBS on NASA Pleiades
+#
+#       module purge
+#       module use /nobackup/lbindle/modulefiles
+#       module load gchp-env/2021.06-gnu
+#
+#    Example 2: LSF on WashU Compute 1 (uses docker containers)
+#
+#       . /spack/share/spack/setup-env.sh
+#       module load openmpi-4.0.1-gcc-9-sdj47y5
+#
+
+#################################################################
+#
+# ADDITIONAL PRE-RUN CONFIGURATION
+#
+# If a subsequent command fails, treat it as fatal (don't continue)
+set -e
+
+# For remainder of script, echo commands to the job's log file
+set -x
+
+# Unlimit resources to prevent OS killing GCHP due to resource usage/
+# Alternatively you can put this in your environment file.
+ulimit -c 0                  # coredumpsize
+ulimit -l unlimited          # memorylocked
+ulimit -u 50000              # maxproc
+ulimit -v unlimited          # vmemoryuse
+ulimit -s unlimited          # stacksize
+
+#################################################################
+#
+# PRE-RUN COMMANDS
+#
+# Print loaded modules
+module list     
+
+# Define log name to include simulation start date
+start_str=$(echo $(cat cap_restart) | sed 's/ /_/g')
+log=gchp.${start_str}z.log
+
+# Update config files, set restart symlink, and do sanity checks
+source setCommonRunSettings.sh
+source setRestartLink.sh
+source checkRunSettings.sh
+
+#################################################################
+#
+# LAUNCH GCHP - 3 examples
+#
+# There are several commands that may work to execute GCHP,
+# and the provided examples may not be appropriate for your system.
+# Check with your system administrator or online documentation about
+# the appropriate way to submit jobs for your scheduler and MPI.
+#
+# Example 1: PBS
+#
+#    mpiexec -n 48 ./gchp > runlog.txt   
+#
+# Example 2: LSF
+#
+#    mpiexec -n 72 ./gchp > runlog.txt
+#
+# Example 3: SLURM
+#
+#    srun -n 48 -N 2 -m plane=24 --mpi=pmix ./gchp
+#
+
+#################################################################
+#
+# POST-RUN COMMANDS
+#
+# If new start time in cap_restart is okay, rename and move restart file
+# and update restart symlink
+new_start_str=$(echo $(cat cap_restart) | sed 's/ /_/g')
+if [[ "${new_start_str}" = "${start_str}" || "${new_start_str}" = "" ]]; then
+   echo "ERROR: cap_restart either did not change or is empty."
+   exit 1
+else
+    N=$(grep "CS_RES=" setCommonRunSettings.sh | cut -c 8- | xargs )    
```

Comment:
@lizziel Does the empty `xargs` do anything? Not seen that pattern before if so.

Reply:
I used xargs here to remove whitespace if any.

Reply:
Let me know if you have a better way.

Reply:
No that's cool, didn't think about using it that way and it's shorter than what I would do (`tr` or `sed`)!